cvat-ui/src/actions/projects-actions.ts

```diff
@@ -0,0 +1,308 @@
+    UPDATE_PROJECT_FAILED = 'UPDATE_PROJECT_FAILED',
+    DELETE_PROJECT = 'DELETE_PROJECT',
+    DELETE_PROJECT_SUCCESS = 'DELETE_PROJECT_SUCCESS',
+    DELETE_PROJECT_FAILED = 'DELETE_PROJECT_FAILED',
+}
+
+// prettier-ignore
+const projectActions = {
+    getProjects: () => createAction(ProjectsActionTypes.GET_PROJECTS),
+    getProjectsSuccess: (array: any[], count: number) => (
+        createAction(ProjectsActionTypes.GET_PROJECTS_SUCCESS, { array, count })
+    ),
+    getProjectsFailed: (error: any) => createAction(ProjectsActionTypes.GET_PROJECTS_FAILED, { error }),
+    updateProjectsGettingQuery: (query: Partial<ProjectsQuery>) => (
+        createAction(ProjectsActionTypes.UPDATE_PROJECTS_GETTING_QUERY, { query })
+    ),
+    createProjects: () => createAction(ProjectsActionTypes.CREATE_PROJECT),
+    createProjectsSuccess: (projectId: number) => (
+        createAction(ProjectsActionTypes.CREATE_PROJECT_SUCCESS, { projectId })
+    ),
+    createProjectsFailed: (error: any) => createAction(ProjectsActionTypes.CREATE_PROJECT_FAILED, { error }),
+    updateProjects: () => createAction(ProjectsActionTypes.UPDATE_PROJECT),
+    updateProjectsSuccess: (project: any) => createAction(ProjectsActionTypes.UPDATE_PROJECT_SUCCESS, { project }),
+    updateProjectsFailed: (project: any, error: any) => (
+        createAction(ProjectsActionTypes.UPDATE_PROJECT_FAILED, { project, error })
+    ),
+    deleteProjects: (projectId: number) => createAction(ProjectsActionTypes.DELETE_PROJECT, { projectId }),
+    deleteProjectsSuccess: (projectId: number) => (
+        createAction(ProjectsActionTypes.DELETE_PROJECT_SUCCESS, { projectId })
+    ),
+    deleteProjectsFailed: (projectId: number, error: any) => (
+        createAction(ProjectsActionTypes.DELETE_PROJECT_FAILED, { projectId, error })
+    ),
+};
+
+export type ProjectActions = ActionUnion<typeof projectActions>;
+
+function updateProjectsGettingQuery(query: Partial<ProjectsQuery>): AnyAction {
+    const action = {
+        type: ProjectsActionTypes.UPDATE_PROJECTS_GETTING_QUERY,
+        payload: {
+            query,
+        },
+    };
+
+    return action;
+}
+
+function getProjects(): AnyAction {
+    const action = {
+        type: ProjectsActionTypes.GET_PROJECTS,
+        payload: {},
+    };
+
+    return action;
+}
+
+function getProjectsSuccess(array: any[], count: number): AnyAction {
+    const action = {
+        type: ProjectsActionTypes.GET_PROJECTS_SUCCESS,
+        payload: {
+            array,
+            count,
+        },
+    };
+
+    return action;
+}
+
+function getProjectsFailed(error: any): AnyAction {
+    const action = {
+        type: ProjectsActionTypes.GET_PROJECTS_FAILED,
+        payload: {
+            error,
+        },
+    };
+
+    return action;
+}
+
+export function getProjectsAsync(query: Partial<ProjectsQuery>): ThunkAction {
+    return async (dispatch: ActionCreator<Dispatch>): Promise<void> => {
+        dispatch(getProjects());
```

Comment:
dispatch(projectsActions.getProjects())

Reply:
The same comment is applicable to all dispatches in this file

Reply:
@bsekachev, done